### PR TITLE
Update azure_rm_galleryimage.py to allow Hyper-V Generation

### LINKED ISSUE #646: VM generation not supported in azure.azcollection.azure_rm_galleryimage
##### SUMMARY
Will allow for v2 generation of VM's in image gallery 

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
azure.azcollection.azure_rm_galleryimage

##### ADDITIONAL INFORMATION
Used to create an image gallery that supports v2 images
```
-   name: Create or update gallery image
    azure_rm_galleryimage:
        hyperVGeneration: V2  <- this is what id like to add
        resource_group: '{{ clientname }}-rg'
        gallery_name: '{{ clientname }}avdimages'
        name: Windows10MSPrimary-Gold
        location: '{{ location }}'
        os_type: windows
        os_state: generalized
        identifier:
            publisher: '{{ clientname }}Publisher'
            offer: '{{ clientname }}Offer'
            sku: '{{ clientname }}SKU'
        disallowed:
            disk_types: Standard_LRS
        recommended:
            memory:
                max: 64
                min: 8
            v_cpus:
                max: 24
                min: 4
```

### PR DESCRIPTION
**Pull request by ghost**
Added Hyper V Generation

##### SUMMARY
Added Hyper-V Generation which allows different versions of Gallery Image - if there is any syntax stuff I need to change please let me know

Fixes #646

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
Hyper-V Generation to allow v2 which allows your Virtual Machines to boot faster when using this boot option

##### ADDITIONAL INFORMATION

```
-   name: Create or update gallery image
    azure_rm_galleryimage:
        hypervgeneration: V2
        resource_group: '{{ clientname }}-rg'
        gallery_name: '{{ clientname }}avdimages'
        name: Windows10MSPrimary-Gold
        location: '{{ location }}'
        os_type: windows
        os_state: generalized
        identifier:
            publisher: '{{ clientname }}Publisher'
            offer: '{{ clientname }}Offer'
            sku: '{{ clientname }}SKU'
        disallowed:
            disk_types: Standard_LRS
        recommended:
            memory:
                max: 64
                min: 8
            v_cpus:
                max: 24
                min: 4
```

### CODE REVIEW

File: plugins/modules/azure_rm_galleryimage.py [hunk 1]
```diff
@@ -69,6 +69,12 @@
             - specialized
         required: true
         type: str
+    hyperveneration:
+        description:
+            - This property allows you to specify the Hyper V Version of the Virtual Machines.
+        choices:
+            - V1
+            - v2
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_galleryimage.py]:
```suggestion
            - V2
```

File: plugins/modules/azure_rm_galleryimage.py [hunk 2]
```diff
@@ -69,6 +69,12 @@
             - specialized
         required: true
         type: str
+    hyperveneration:
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_galleryimage.py]:
```suggestion
    hypervgeneration:
```

**Comment by Fred-sun** [comment]
Kindly ping!

**Comment by Fred-sun** [comment]
small change request!

**Comment by Fred-sun** [comment]
Kindly ping!

**Review by ghost — COMMENTED** [review]
Looks great

File: plugins/modules/azure_rm_galleryimage.py [hunk 3]
```diff
@@ -69,6 +69,12 @@
             - specialized
         required: true
         type: str
+    hypervgeneration:
+        description:
+            - This property allows you to specify the Hyper V Version of the Virtual Machines.
+        choices:
+            - V1
+            - V2
```
Comment (xuzhang3) [inline plugins/modules/azure_rm_galleryimage.py:77]: `hypervgeneration` should specify  type `str` as code defined

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**